woocommerce/woocommerce
# Fix test in Admin reports for HPOS compat

### All Submissions:






### Changes proposed in this Pull Request:



Closes #36678 
Targets #36910 

Modifies [Admin orders API](https://github.com/woocommerce/woocommerce/pull/36946/files#diff-49b20224dc7c1786cf00f3622403ede4f67adbe7d4b08047a92ba6254df8e8dbR80) to also support HPOS in partial order id matching query by selecting correct table based on whether HPOS is enabled or not.

Also marks several test as skipped for legacy reports which we don't plan to support for HPOS.

Additionally, modifies the test `test_setting_default_address_fields` so that it's data provider returns a closure instead of the DB object. The reason for this is that all the data providers are evaluated at once by PHPUnit, which means anything adding data to DB may corrupt the results of all other tests. 



### How to test the changes in this Pull Request:




The code is adequately tested by unit tests so no specific testing is needed. Alternatively,

1. Send an API request for a partial order number like so: `wc-analytics/orders?number=1` and verify that IDs for all returned orders starts with `1`.



### Other information:




### FOR PR REVIEWER ONLY:
